File: Factory Pattern/Cpp/Pizza/NYStylePizza/NYStyleCheesePizza.cpp
#include "../Pizza.cpp"

class NYStyleCheesePizza : public Pizza {
public:
    NYStyleCheesePizza() {
        name = "뉴욕 스타일 소스와 치즈 피자";
        dough = "씬 크러스트 도우";
        sauce = "마리나라 소스";

        toppings.push_back("\t잘게 썬 레지아노 치즈\n");
    }
};

File: Patterns/Factory Method Pattern/CSharp/Pizza/NYStylePizza/NYStyleCheesePizza.cs
using System;

namespace Pizza.NYStylePizza
{
    public class NYStyleCheesePizza : Pizza 
    {
        public NYStyleCheesePizza() :
            base("뉴욕 스타일 소스와 치즈 피자", "씬 크러스트 도우", "마리나라 소스")
        {
            toppings.Add("\t잘게 썬 레지아노 치즈");
        }
    }
}